LibreNMS/Modules/Routes.php

```diff
@@ -0,0 +1,346 @@
+            Log::info('Skipping "ipCidrRouteTable"');
+
+            return new Collection;
+        }
+    }
+
+    private function discoverRfcRoutes(Device $device): Collection
+    {
+        Log::info('RFC1213-MIB::ipRouteTable');
+
+        return SnmpQuery::hideMib()->walk(['RFC1213-MIB::ipRouteTable'])
+        ->mapTable(function ($data) use ($device) {
+            return new Route([
+                'port_id' => PortCache::getIdFromIfIndex($data['ipRouteIfIndex'] ?? 0, $device->device_id) ?? 0,
+                'context_name' => '',
+                'inetCidrRouteType' => $data['ipRouteType'] ?? 0,
+                'inetCidrRouteProto' => $data['ipRouteProto'] ?? 0,
+                'inetCidrRouteIfIndex' => $data['ipRouteIfIndex'] ?? 0,
+                'inetCidrRouteNextHopAS' => '0',
+                'inetCidrRouteMetric1' => $data['ipRouteMetric1'] ?? 0,
+                'inetCidrRouteDestType' => 'ipv4',
+                'inetCidrRouteDest' => $data['ipRouteDest'] ?? '',
+                'inetCidrRouteNextHopType' => 'ipv4',
+                'inetCidrRouteNextHop' => $data['ipRouteNextHop'] ?? '',
+                'inetCidrRoutePfxLen' => $data['ipRouteMask'] ?? '',
+                'inetCidrRoutePolicy' => $data['ipRouteInfo'] ?? '',
+            ]);
+        })->filter();
+    }
+
+    private function discoverIpv6MibRoutes(Device $device): Collection
+    {
+        Log::info('IPV6-MIB::ipv6RouteTable');
+
+        return SnmpQuery::hideMib()->walk(['IPV6-MIB::ipv6RouteTable'])
+        ->mapTable(function ($data, $dst = '', $pfxLen = '', $tos = '') use ($device) {
+            return new Route([
+                'port_id' => PortCache::getIdFromIfIndex($data['ipv6RouteIfIndex'] ?? 0, $device->device_id) ?? 0,
+                'context_name' => '',
+                'inetCidrRouteIfIndex' => $data['ipv6RouteIfIndex'] ?? 0,
+                'inetCidrRouteType' => $data['ipv6RouteType'] ?? 0,
+                'inetCidrRouteProto' => $data['ipv6RouteProtocol'] ?? 0,
+                'inetCidrRouteNextHopAS' => '0',
+                'inetCidrRouteMetric1' => $data['ipv6RouteMetric'] ?? 0,
+                'inetCidrRouteDestType' => 'ipv6',
+                'inetCidrRouteDest' => $dst,
+                'inetCidrRouteNextHopType' => 'ipv6',
+                'inetCidrRouteNextHop' => $data['ipv6RouteNextHop'] ?? '',
+                'inetCidrRoutePolicy' => $data['ipv6RoutePolicy'] ?? '',
+                'inetCidrRoutePfxLen' => $pfxLen,
+            ]);
+        })->filter();
+    }
+
+    private function discoverVpnVrfRoutes(Device $device, $max_routes): Collection
+    {
+        Log::info('MPLS-L3VPN-STD-MIB');
+
+        foreach (SnmpQuery::hideMib()->walk(['MPLS-L3VPN-STD-MIB::mplsL3VpnVrfPerfCurrNumRoutes'])->table(1) as $vpnId => $data) {
+            //mes24xx
+            if (empty($data['mplsL3VpnVrfPerfCurrNumRoutes'])) {
+                Log::info('Skipping all MPLS routes because invalid MPLS-L3VPN-STD-MIB data');
```

Comment:
If one single VRF has 0 routes, then all routes for this devices are dropped. This is not what you meant I suppose. This breaks a few of my MPLS routers route discovery

Reply:
> This is not what you meant I suppose. T

yes, you are right, i wanted to escape bad data from mes24xx :slightly_frowning_face: 
will rewrite soon

Reply:
done